[CS2113-T16-3] Expiry Eliminator

bernardboey opened the pull request:
Expiry Eliminator is a desktop app for managing the freshness of ingredients in your kitchen and the recipes you want to cook, optimized for use via a Command Line Interface (CLI). If you are a young adult who is living in your own home and you can type fast, Expiry Eliminator can help you manage your ingredients and recipes quickly.

vincentlauhl committed:
Merge pull request #65 from vincentlauhl/BugfixCookedRecipe

Bugfix cooked recipe

vincentlauhl committed:
Sequence diagram for AddRecipe.

vincentlauhl committed:
Sequence diagram for CookedRecipeCommand

vincentlauhl committed:
Sequence diagram for DeleteRecipeCommand

vincentlauhl committed:
Sequence diagram for ListRecipesUserCanCookCommand

JoshHDs committed:
Add .puml diagram for List command
Add.puml diagram for ListIngredientsExpired command

JoshHDs committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113-T16-3/tp

JoshHDs committed:
Merge branch 'master' into sequenceDiagram-branch

vincentlauhl committed:
Change Model to Data

vincentlauhl committed:
Component diagram for Data

vincentlauhl committed:
Update Developer guide for AddRecipe, DeleteRecipe, ListRecipeUserCanCook, CookedRecipe, and Data

JoshHDs committed:
Add sequence diagram for DeleteExpiredIngredientsCommand.
Add sequence diagram for ShoppingListCommand.
Add sequence diagram for ViewIngredientCommand.
Add sequence diagram for UpdateUnitsComand.

JoshHDs committed:
Merge pull request #67 from vincentlauhl/DGUpdate2

Dg update2

JoshHDs committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113-T16-3/tp

JoshHDs committed:
Merge branch 'master' into sequenceDiagram-branch

vincentlauhl committed:
Update ListRecipesUserCanCookCommand such that it informs the user if the ingredients are expired for the recipe they can cook

vincentlauhl committed:
Minor update to CookedRecipeCommand such that it updates the Ingredient Repository saved to the file.

vincentlauhl committed:
Minor update to prepareCookedRecipe so that it displays the correct error message

vincentlauhl committed:
Add new test for ListRecipesUserCanCookCommand

JoshHDs committed:
Merge pull request #68 from vincentlauhl/CheckExpiryDateForRecipesUserCanCook

Check expiry date for recipes user can cook

JoshHDs committed:
Update sequence diagram for DeleteExpiredIngredients
Update  sequence diagram for List
Update  sequence diagram for ListIngredientsExpired
Update sequence diagram for ShoppingList
Update  sequence diagram for UpdateUnits
Update sequence diagram for ViewIngredient

JoshHDs committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113-T16-3/tp

JoshHDs committed:
Merge branch 'master' into sequenceDiagram-branch

JoshHDs committed:
Update DG

JoshHDs committed:
Merge pull request #69 from JoshHDs/sequenceDiagram-branch

Sequence diagram branch

JoshHDs committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113-T16-3/tp

Alvinlj00 commented on docs/DeveloperGuide.md:
```diff
@@ -4,15 +4,170 @@
+## Implementation
+This section describes how the features are implemented.
+
+### List all Ingredients Feature
```
```suggestion
### List All Ingredients Feature
```

Alvinlj00 commented on docs/DeveloperGuide.md:
```diff
@@ -4,15 +4,170 @@
+
+<br/>
+
+###List Recipes User Can Cook Feature
```
```suggestion
### List Recipes User Can Cook Feature
```

Alvinlj00 commented on docs/DeveloperGuide.md:
```diff
@@ -4,15 +4,170 @@
+
+### List Expired Ingredients Feature
+
+The list expired ingredients feature is performed by `IngredientRepository`. It loops through all the
+ingredient storages. For each ingredient storage, it gets the type of ingredient being stored in that 
+storage, and creates an empty storage for that ingredient type to store the batches of that ingredient 
+that are expired. The code then loops through all the batches of that ingredients, and add the expired 
+batches to the expire ingredient storage. The expired ingredient storage is converted into a string and
+is shown as the output. Below is the sequence diagram for how list expired ingredients works.
```
Would it be clearer if this was in point form instead?

Alvinlj00 commented on docs/DeveloperGuide.md:
```diff
@@ -4,15 +4,170 @@
+
+
+
+## Implementation
```
Could a table of commands be added so that navigation through different commands would be easier?

Alvinlj00 commented on docs/DeveloperGuide.md:
```diff
@@ -4,15 +4,170 @@
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
-## Design & implementation
+## Design
```
Would you consider including some class or object diagrams to illustrate how these components work?

Alvinlj00 reviewed (commented):
Overall, it is a well-structured DG with clear headers and diagrams within each section. It might be better if, for each feature, the explanation is less wordy and in point form or steps.

KairuiHu committed:
Add architecture diagram for storage

vincentlauhl committed:
Update code so that it saves the ingredients and recipes to the text file each time the command executes.

vincentlauhl committed:
Change ListRecipesUserCanCookCommand so that it is consistent with ListIngredientExpiredCommand.

KairuiHu committed:
Merge pull request #73 from vincentlauhl/BugFix

Bug fix

KairuiHu committed:
Reformat output in txt file to remove extra whitespace, and remove redundant saveData methods

KairuiHu committed:
Merge branch 'master' into branch-architecture

JoshHDs committed:
Update help command
Add units deleted message

JoshHDs committed:
Merge pull request #72 from KairuiHu/branch-architecture

Add architecture diagram for storage

JoshHDs committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113-T16-3/tp

JoshHDs committed:
Merge branch 'master' into bugFix-branch

JoshHDs committed:
Fix shopping list no units bug

bernardboey committed:
Remove unnecessary imports and clean up code

KairuiHu committed:
Modify code for LoadData to load ingredients and recipes with no units

KairuiHu committed:
Improve code quality

JoshHDs committed:
Fix test code

bernardboey committed:
Merge pull request #76 from bernardboey/clean-up-code

Remove unnecessary imports and clean up code

JoshHDs committed:
Merge pull request #77 from KairuiHu/branch-modifyLoadData

Branch modify load data

JoshHDs committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113-T16-3/tp

JoshHDs committed:
Merge branch 'master' into bugFix-branch

JoshHDs committed:
Merge pull request #75 from JoshHDs/bugFix-branch

Bug fix branch

bernardboey committed:
Add class diagram for Parser

bernardboey committed:
Add class diagram for parser to the developer guide

bernardboey committed:
Prevent deletion of ingredient if it is used in a recipe

bernardboey committed:
Polish usage messages

bernardboey committed:
Update expected test output

bernardboey committed:
Merge pull request #78 from bernardboey/v2.0

Add class diagram for parser, and make minor fixes to code

vincentlauhl committed:
Change code such that it outputs the correct error message for UpdateUnits

vincentlauhl committed:
Merge pull request #79 from vincentlauhl/CorrectErrorMessageUpdateUnits

Correct error message for UpdateUnitsCommand

bernardboey committed:
Fix delete ingredient bug

bernardboey committed:
Merge pull request #80 from bernardboey/v2.0-fix-bugs

Fix delete ingredient bug

bernardboey committed:
Remove TODO comments

bernardboey committed:
Merge pull request #81 from bernardboey/v2.0-fix-bugs-2

Remove TODO comments

JoshHDs committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113-T16-3/tp

vincentlauhl committed:
Change version of our app

vincentlauhl committed:
Add a few lines to UG

bernardboey committed:
Update user guide

vincentlauhl committed:
Merge pull request #82 from vincentlauhl/UGUpdate1

Ug update1

bernardboey committed:
Merge pull request #83 from bernardboey/user-guide-1

Update user guide

williamlamjy commented on docs/DeveloperGuide.md:
```diff
@@ -4,15 +4,179 @@
+`ExpiryDateParser`, `UnitParser`, `IngredientParser`, `QuantityParser`, `RecipeParser`. The dependencies have been 
+omitted from the diagram for simplicity.
+
+![](diagrams/ParserClassDiagram.png)
```
Can increase the width of this diagram as the diagram is very small. Could also separate the two diagrams instead of putting it side by side

williamlamjy commented on docs/diagrams/List.puml:
```diff
@@ -0,0 +1,34 @@
+    activate ingredientRepository
+
+        loop ingredientStorages
+            ingredientRepository -> ingredientStorage
```
could add a note for concatenate here. Do u not have a method to include here for specific concatenation? this is for step 4

williamlamjy commented on docs/diagrams/ListIngredientsExpired.puml:
```diff
@@ -0,0 +1,53 @@
+
+ListIngredientsExpiredCommand -> ingredientRepository: findExpiredIngredients()
+activate ingredientRepository
+
```
is this part necessary because u had the overview earlier. Maybe u could consider removing the parser and ListIngredientsExpiredCommand to make ur diagram smaller because right now there is too much content on ur seq diagram and the content that hasnt been shown yet is really small which is the interaction between ingredientRepository and IngredientStorage and expiredIngredientStorage

williamlamjy replied:
maybe u could have a separate diagram for parser and the corresponding commands since the same procedure would be happening for many commands. this could be a generalised diagram so that u can omit the parser to command portions in most of ur diagrams. This would help limit the scope of ur diagram

jerrelllzw commented on docs/DeveloperGuide.md:
```diff
@@ -4,15 +4,179 @@
+### Data
+**API** : [Data.java](https://github.com/AY2122S1-CS2113-T16-3/tp/tree/master/src/main/java/expiryeliminator/data)
 
+![](diagrams/Data.png)
```
Can consider adding a diagram showing the overall architecture of the application (i.e. include UI and storage components) to give readers a bigger picture.

jerrelllzw commented on docs/DeveloperGuide.md:
```diff
@@ -4,15 +4,179 @@
+ingredient storages, and concatenates them into a string which is shown as the output list. Below is the
+sequence diagram for how list works.
+
+![](diagrams/List.png)
```
Diagram is very clean and easy to understand, good job!

williamlamjy commented on docs/diagrams/AddRecipe.puml:
```diff
@@ -0,0 +1,45 @@
+@startuml
```
this diagram is really big maybe u could split it into one logic diagram that calls the functions in a data diagram that only has the recipe class. then add another diagram just for the data diagram. the wording is way too small on this diagram

williamlamjy commented on docs/diagrams/CookedRecipe.puml:
```diff
@@ -0,0 +1,50 @@
+@startuml
```
same as the comment above maybe split the logic and data diagrams

williamlamjy reviewed (commented):
can consider adding a Styles.puml file to format all ur diagrams with colours and set font colour, sizing and font type. Also can hide the footbox on all seq diagrams by doing so in this styles file. A lot of the wording in the diagrams are way too small so setting font size in this file might be useful too.

jerrelllzw commented on docs/DeveloperGuide.md:
```diff
@@ -4,15 +4,179 @@
 ### Target user profile
 
-{Describe the target user profile}
+Young adults who are living in their own home.
```
Can consider being more specific. e.g. young adults who cook often and needs to keep track of their ingredients.

jerrelllzw commented on docs/DeveloperGuide.md:
```diff
@@ -4,15 +4,179 @@
+ - Each `Recipe` has its ingredients and respective quantities stored in `IngredientQuantity`
+ - The `Ingredient Storage` separates a specific ingredient into batches of quantities, 
+   according to the expiry date.
+ - `Ingredient` stores the information of the ingredients, which are its name and unit, 
```
Can this line be phrased better?

vincentlauhl replied:
May I know which part is causing you confusion, so that I can phrase it properly accordingly.

Thanks!

bernardboey committed:
Add explanation of Parser to the developer guide

bernardboey committed:
Add sequence diagram for Parser

bernardboey committed:
Merge pull request #84 from bernardboey/developer-guide-2

Add explanation of Parser to the developer guide

vincentlauhl committed:
Add features in user guide and its comand summaries.

bernardboey committed:
Update docs/DeveloperGuide.md

Co-authored-by: Alvinlj00 <69453924+Alvinlj00@users.noreply.github.com>

bernardboey committed:
Update docs/DeveloperGuide.md

Co-authored-by: Alvinlj00 <69453924+Alvinlj00@users.noreply.github.com>

bernardboey committed:
Rename ArgParser class to ArgsParser

bernardboey committed:
Add style.puml

bernardboey committed:
Add simplified class diagram for parser

bernardboey committed:
Extract arg parsers into separate class diagram

bernardboey committed:
Modify class diagram for parser

bernardboey committed:
Modify sequence diagram for parser

bernardboey committed:
Update developer guide

bernardboey committed:
Clean up markdown files

bernardboey committed:
Merge pull request #86 from vincentlauhl/UGUpdate2

Add features in user guide and its comand summaries.

bernardboey committed:
Merge pull request #87 from bernardboey/developer-guide-3

Update developer guide

JoshHDs committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113-T16-3/tp

JoshHDs committed:
Update documentation

JoshHDs committed:
Merge pull request #88 from JoshHDs/UGV2-branch

Ugv2 branch

jyxhazcake commented on docs/DeveloperGuide.md:
```diff
@@ -2,32 +2,248 @@
+Here is the sequence diagram for how cooked recipe works if the amount of ingredients in the `Ingredient Repository`
+is sufficient to be deducted.
+
+![](diagrams/CookedRecipe.png)
```
![image](https://user-images.githubusercontent.com/77771132/139371508-c6110d27-99b6-4d53-82f9-532c2837506b.png)
Are the entities shown at the bottom of the diagrams required? All your sequence diagrams above under Implementation have these.

jyxhazcake commented on docs/DeveloperGuide.md:
```diff
@@ -2,32 +2,248 @@
+This is a (partial) sequence diagram of parsing an "add recipe" command. It demonstrates the interaction between the
+`Parser`, `ArgsParser`, and the corresponding single arg and multiple arg parsers (`RecipeParser`, `IngredientParser`, and `QuantityParser`).
+
+![](diagrams/ParserSequenceDiagram.png)
```
![image](https://user-images.githubusercontent.com/77771132/139372761-c0c2b99e-b8cf-4eb4-b402-480bf7b98931.png)
Unclear whether the left-pointing arrows are callbacks or return arrows? If returning they should be dashed instead.

jyxhazcake commented on docs/DeveloperGuide.md:
```diff
@@ -2,32 +2,248 @@
+
+This is a more detailed class diagram of the `Parser`, `ArgsParser`, and `Prefix` classes. Note that `Parser` has 4 single arg prefixes, 3 multiple arg prefixes, and 3 optional arg prefixes. These prefixes correspond to the various arguments that are accepted by the various commands.
+
+![](diagrams/ParserClassDiagram.png)
```
![image](https://user-images.githubusercontent.com/77771132/139373077-904e338e-39ba-4af8-8b6d-98f6d6cc0ff2.png)
There are more details omitted in this class diagram like missing multiplicities. Perhaps you can consider adding them in as well.

jyxhazcake commented on docs/DeveloperGuide.md:
```diff
@@ -2,32 +2,248 @@
+
+### Delete Expired Ingredients Feature
+
+The delete expired ingredients feature is performed by the `IngredientRepository`. It loops through all
```
Perhaps you can consider breaking down the explanation into points instead. Would make it clearer.

jyxhazcake reviewed (commented):
Diagrams are generally really well done, with some discrepancies here and there. Explanations are also generally pretty clear. Good job.

KairuiHu committed:
Add features in User guide

KairuiHu committed:
Merge branch 'master' into branch-userguide

# Conflicts:
#	docs/UserGuide.md

KairuiHu committed:
Merge pull request #89 from KairuiHu/branch-userguide

Branch userguide

bernardboey committed:
Add table of contents to user guide and developer guide

bernardboey committed:
Modify location of table of contents for user guide

bernardboey committed:
Update user guide

bernardboey committed:
Merge pull request #90 from bernardboey/user-guide-2

Update user guide

bernardboey committed:
Update ShoppingListCommand.java

bernardboey committed:
Update AddRecipeCommand.java

bernardboey committed:
Update UpdateRecipeCommand.java

bernardboey committed:
Update UserGuide.md

bernardboey committed:
Update UserGuide.md

bernardboey committed:
Update EXPECTED.TXT

bernardboey committed:
Update UserGuide.md

bernardboey committed:
Update .gitignore to ignore data folder

JoshHDs committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113-T16-3/tp

JoshHDs committed:
Add repo sense compatibility code

JoshHDs committed:
Fix bugs from PED

JoshHDs committed:
Fix bugs from PE-D

JoshHDs committed:
Fix code style issues

JoshHDs committed:
Merge pull request #112 from JoshHDs/BugFix2-branch

Bug fix2 branch

bernardboey committed:
Allow commands to be typed in upper or mixed case

bernardboey committed:
Display error message instead of throwing runtime exception

bernardboey committed:
Improve error message if ingredient already exists

bernardboey committed:
Remove quantity and expiry date from AddIngredientCommand

bernardboey committed:
Modify class diagram for parser to remove outdated prefixes

bernardboey committed:
Prevent recipe and ingredient names from having special digits

bernardboey committed:
Modify IO redirection tests

bernardboey committed:
Merge pull request #113 from bernardboey/v2.1

Fix bugs for parser and add ingredient command

vincentlauhl committed:
Add example outputs to features

vincentlauhl committed:
Add description of add recipe feature.

vincentlauhl committed:
Update add recipe usage message.

vincentlauhl committed:
Update message display format for DeleteRecipeCommandd

vincentlauhl committed:
Change assertion position to allow JUnit test to work properly.

vincentlauhl committed:
Simplify JUnit Test Utils functions and add java docs

vincentlauhl committed:
Minor update to adhere to codestyle

vincentlauhl committed:
Update Expected.txt for tests to pass.

vincentlauhl committed:
Merge pull request #114 from vincentlauhl/UGAndJUnitUpdate

Ug and j unit update

JoshHDs committed:
Update sequence diagrams

JoshHDs committed:
Merge branch 'master' into BugFix3-branch

KairuiHu committed:
Enable add ingredient for recipe later using udpate recipe, implement another feature "delete ingredient in recipe", solve the issue in Tutor's code review

KairuiHu committed:
Improve code quality

KairuiHu committed:
Improve output format to pass the test

KairuiHu committed:
Modify output format  to pass the test

KairuiHu committed:
Change the output format

JoshHDs committed:
Merge pull request #115 from KairuiHu/branch-fixUpdate

Enable add ingredient for recipe later using update recipe, implement…

JoshHDs committed:
Update sequence diagrams

JoshHDs committed:
Merge branch 'master' into BugFix3-branch

# Conflicts:
#	src/main/java/expiryeliminator/parser/Parser.java

JoshHDs committed:
Merge pull request #116 from JoshHDs/BugFix3-branch

Bug fix3 branch

JoshHDs committed:
Update user guide with output examples

JoshHDs committed:
Merge pull request #117 from JoshHDs/BugFix4-branch

Update user guide with output examples

bernardboey committed:
Update bernardboey.md

bernardboey committed:
Update bernardboey.md

bernardboey committed:
Update UserGuide.md

vincentlauhl committed:
Updates to UML diagrams for implemented features and design

vincentlauhl committed:
Update PPP

vincentlauhl committed:
Add authorship to AddRecipeCommand

vincentlauhl committed:
Add authorship to methods in Recipe

KairuiHu committed:
Use update recipe command to delete ingredients

vincentlauhl committed:
Add authorship to Parser.

KairuiHu committed:
Improve code quality

KairuiHu committed:
Merge pull request #118 from KairuiHu/branch-CodeQuality

Use update recipe command to delete ingredients

vincentlauhl committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113-T16-3/tp into DGUpdate3

# Conflicts:
#	src/main/java/expiryeliminator/parser/Parser.java

vincentlauhl committed:
Merge pull request #119 from vincentlauhl/DGUpdate3

Dg update3

JoshHDs committed:
Add manual testing section in the DG

JoshHDs committed:
Add Joshua's PPP
Add Logic Diagram

bernardboey committed:
Modify IllegalValueException to allow storing of error message

bernardboey committed:
Remove unused code

bernardboey committed:
Add back assertion for removing ingredient batch based on expiry

bernardboey committed:
Prevent decrementing by zero quantity

bernardboey committed:
Prevent incrementing by zero quantity

Modify IngredientStorage#add(quantity, expiryDate) method to
throw an exception if quantity is less than or equal to zero

bernardboey committed:
Rename Recipe#add() to Recipe#addIngredient()

bernardboey committed:
Modify DuplicateDataException to allow storing of error message

vincentlauhl committed:
Update Instructions for manual testing and some minor changes to DG

vincentlauhl committed:
Add interactions of data component and update puml diagram.

vincentlauhl committed:
Add PPP and a skeleton file for teammate.

bernardboey committed:
Fix typo in DG

JoshHDs committed:
Update Logic diagram

bernardboey committed:
Modify OptionalArgPrefix constructor to use SingleArgPrefix as input

bernardboey committed:
Add PREFIX_UNIT and also trim command word(s)

bernardboey committed:
Add assertion to Ingredient#setUnit(unit)

bernardboey committed:
Extract error messages as variables

bernardboey committed:
Modify ArgsParser to accept blank arguments

Also reject arguments that have a space immediately after prefix
Also modify tests to use more precise error messages

JoshHDs committed:
Merge pull request #122 from JoshHDs/BugFix5-branch

Bug fix5 branch

vincentlauhl committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113-T16-3/tp into DGUpdate4

# Conflicts:
#	docs/DeveloperGuide.md

bernardboey committed:
Merge pull request #120 from vincentlauhl/DGUpdate4

Dg update4

JoshHDs committed:
Merge pull request #124 from bernardboey/v2.1-fix-bugs

Disallow incrementing/decrementing by zero quantity and spaces after prefixes

KairuiHu committed:
Add ppp and update manual testing for DG

KairuiHu committed:
Add new line to pass the checkstyle test

KairuiHu committed:
Merge pull request #125 from KairuiHu/branch-Document

Add ppp and update manual testing for DG

vincentlauhl committed:
Add logging for features

vincentlauhl committed:
Add method to check if arraylist has duplicate ingredients and some refactoring.

vincentlauhl committed:
Change position for expected and actual

vincentlauhl committed:
Update code to adhere to coding standards.

vincentlauhl committed:
Change method on checking for same ingredients.

JoshHDs committed:
Merge pull request #126 from vincentlauhl/BugFixAndLogging

Bug fix and logging

JoshHDs committed:
Update sequence diagram
Update Joshua's PPP

bernardboey committed:
Update AboutUs.md

JoshHDs committed:
Edit About us

JoshHDs committed:
Merge branch 'master' into BugFix6-branch

# Conflicts:
#	docs/AboutUs.md

JoshHDs committed:
Merge pull request #127 from JoshHDs/BugFix6-branch

Bug fix6 branch

JoshHDs committed:
Fix bug in Joshhds PPP

JoshHDs committed:
Merge pull request #128 from JoshHDs/BugFix7-branch

Fix bug in Joshhds PPP

JoshHDs committed:
Fix bug in Joshhds PPP

vincentlauhl committed:
Update AboutUs to show Github page and PPP

JoshHDs committed:
Merge pull request #129 from JoshHDs/BugFix8-branch

Fix bug in Joshhds PPP

vincentlauhl committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113-T16-3/tp into PPPUpdate

vincentlauhl committed:
Merge pull request #130 from vincentlauhl/PPPUpdate

Ppp update

vincentlauhl committed:
Update PPP to keep to page limit

vincentlauhl committed:
Merge pull request #131 from vincentlauhl/PPPBugFix

Update PPP to keep to page limit

vincentlauhl committed:
Update PPP to adhere to 2 page limit

vincentlauhl committed:
Merge pull request #132 from vincentlauhl/PPPBugFix2

Update PPP to adhere to 2 page limit

vincentlauhl committed:
Update PPP to adhere to 2 page limit

vincentlauhl committed:
Merge pull request #133 from vincentlauhl/PPPBugFix3

Update PPP to adhere to 2 page limit

KairuiHu committed:
Update DG and PPP

KairuiHu committed:
Merge pull request #134 from KairuiHu/branch-updateDG

Update DG and PPP

JoshHDs committed:
Add Architecture Diagram

JoshHDs committed:
Merge branch 'master' into BugFix9-branch

JoshHDs committed:
Merge pull request #135 from JoshHDs/BugFix9-branch

Bug fix9 branch

KairuiHu committed:
Add new digrams and update UserGuide

KairuiHu committed:
Merge pull request #136 from KairuiHu/branch-updateUG

Add new digrams and update UserGuide

bernardboey committed:
Remove unused file

bernardboey committed:
Update PPP

bernardboey committed:
Clean up and add more links to PPP

bernardboey committed:
Merge pull request #137 from bernardboey/ppp-1

Update Bernard's PPP

KairuiHu committed:
Add sequence diagram to the DG

KairuiHu committed:
Merge pull request #139 from KairuiHu/branch-AddNewDiagrams

Add sequence diagram to the DG

bernardboey committed:
Update main README.md

bernardboey committed:
Clean up AboutUs.md

bernardboey committed:
Change name to full name

KairuiHu committed:
Improve grammar

KairuiHu committed:
Merge pull request #140 from KairuiHu/branch-Format

Improve grammar

KairuiHu committed:
Make the ppp more concise

KairuiHu committed:
Merge pull request #141 from KairuiHu/branch-ppp

Make the ppp more concise

KairuiHu committed:
reformat the ppp

KairuiHu committed:
Merge pull request #142 from KairuiHu/branch-ppp1

reformat the ppp

bernardboey committed:
Update user guide

bernardboey committed:
Add getFormattedUnit() method to IngredientStorage and Ingredient

bernardboey committed:
Display units in IncrementCommand and DecrementCommand

vincentlauhl committed:
Minor update to manual testing for launch.

bernardboey committed:
Update I/O redirection test

vincentlauhl committed:
Merge pull request #144 from vincentlauhl/MinorUgUpdate

Minor update to manual testing for launch.

bernardboey committed:
Merge branch 'master' into documentation-1

# Conflicts:
#	docs/UserGuide.md

bernardboey committed:
Merge pull request #143 from bernardboey/documentation-1

Update user guide and other documentation

KairuiHu committed:
Add authorship

KairuiHu committed:
Merge pull request #145 from KairuiHu/branch-capitalization

Add authorship

bernardboey committed:
Update user guide

JoshHDs committed:
Fix spelling errors in DG

JoshHDs committed:
Merge pull request #146 from JoshHDs/BugFix10-branch

Fix spelling errors in DG

vincentlauhl committed:
Update to Instructions for Manual Testing for clarity

vincentlauhl committed:
Minor update to PPP.

vincentlauhl committed:
Merge pull request #147 from vincentlauhl/FinalDGUpdate

Update to Instructions for Manual Testing for clarity

vincentlauhl committed:
Update PPP

vincentlauhl committed:
Merge pull request #148 from vincentlauhl/PPPFinalUpdate

Update PPP

bernardboey committed:
Update ParserClassDiagram

bernardboey committed:
Update developer guide

bernardboey committed:
Update ParserClassDiagram

bernardboey committed:
Merge pull request #150 from bernardboey/v2.1-2

Update documentation

bernardboey committed:
Add JavaDoc comments for Parser

bernardboey committed:
Add JavaDoc comments for ArgsParser

bernardboey committed:
Add more logging

bernardboey committed:
Add more tests

bernardboey committed:
Merge pull request #151 from bernardboey/v2.1-3

Add JavaDoc comments and more tests

bernardboey committed:
Update bernardboey.md

bernardboey committed:
Update archiveClassifier version

bernardboey committed:
Update bernardboey.md

bernardboey committed:
Update AddRecipeCommandTest.java
